sessions/test_js_challenge.py

```diff
@@ -706,3 +706,111 @@ def test_cookie_expires(self):
             self.klog.find("http_sess: sticky cookie value expired", cond=dmesg.amount_equals(1)),
             1,
         )
+
+@parameterize_class(
+    [
+        {"name": "Http", "clients": [DEPROXY_CLIENT]},
+        {"name": "H2", "clients": [DEPROXY_CLIENT_H2]},
+    ]
+)
+class JSChallengeWithCustomRules(BaseJSChallenge):
+    backends = [
+        {
+            "id": "server-1",
+            "type": "deproxy",
+            "port": "8000",
+            "response": "static",
+            "response_content": (
+                "HTTP/1.1 200 OK\r\n"
+                + f"Date: {HttpMessage.date_time_string()}\r\n"
+                + "Server: deproxy\r\n"
+                + "Content-Length: 0\r\n\r\n"
+            ),
+        },
+    ]
+
+    tempesta = {
+        "config": f"""
+
+        listen 80;
+        listen 443 proto=h2;
+
+        srv_group default {{
+            server ${{server_ip}}:8000;
+        }}
+
+        tls_certificate ${{tempesta_workdir}}/tempesta.crt;
+        tls_certificate_key ${{tempesta_workdir}}/tempesta.key;
+        tls_match_any_server_name;
+
+        block_action attack reply;
+        block_action error reply;
+
+        cache 2;
+        cache_fulfill * *;
+
+        sticky {{
+            cookie enforce name=cname max_misses={MAX_MISSES};
+            js_challenge resp_code=503 delay_min={DELAY_MIN} delay_range={DELAY_RANGE}
+                        ${{tempesta_workdir}}/js1.html;
+            sess_lifetime 3;
+        }}
+
+        vhost default {{
+            tls_match_any_server_name;
+            proxy_pass default;
+        }}
+
+        http_chain {{
+            uri == "*1.html" -> jsch;
+            -> default;
+        }}
+
+        """
+    }
+
+    def setUp(self):
+        super().setUp()
+        self.klog = dmesg.DmesgFinder(disable_ratelimit=True)
+        self.assert_msg = "Expected nums of warnings in `journalctl`: {exp}, but got {got}"
+        # Cleanup part
+        self.addCleanup(self.cleanup_klog)
+
+    def cleanup_klog(self):
+        if hasattr(self, "klog"):
+            del self.klog
```

Comment:
You can use [self.oops](https://github.com/tempesta-tech/tempesta-test/blob/c1bb410e1dba70c9ba0ae86eb0c9b05e48f6483e/framework/tester.py#L344) variable here.